pkg/apis/coordination/validation/validation.go

```diff
@@ -50,3 +51,32 @@ func ValidateLeaseSpec(spec *coordination.LeaseSpec, fldPath *field.Path) field.
 	}
 	return allErrs
 }
+
+// ValidateLeaseCandidate validates a LeaseCandidate.
+func ValidateLeaseCandidate(lease *coordination.LeaseCandidate) field.ErrorList {
+	allErrs := validation.ValidateObjectMeta(&lease.ObjectMeta, true, ValidLeaseCandidateName, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateLeaseCandidateSpec(&lease.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+func ValidLeaseCandidateName(name string, prefix bool) []string {
+	return utilvalidation.IsConfigMapKey(name)
+}
+
+// ValidateLeaseCandidateUpdate validates an update of LeaseCandidate object.
+func ValidateLeaseCandidateUpdate(lease, oldLeaseCandidate *coordination.LeaseCandidate) field.ErrorList {
+	allErrs := validation.ValidateObjectMetaUpdate(&lease.ObjectMeta, &oldLeaseCandidate.ObjectMeta, field.NewPath("metadata"))
+	allErrs = append(allErrs, ValidateLeaseCandidateSpec(&lease.Spec, field.NewPath("spec"))...)
+	return allErrs
+}
+
+// ValidateLeaseCandidateSpec validates spec of LeaseCandidate.
+func ValidateLeaseCandidateSpec(spec *coordination.LeaseCandidateSpec, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	if spec.LeaseDurationSeconds != nil && *spec.LeaseDurationSeconds <= 0 {
+		fld := fldPath.Child("leaseDurationSeconds")
+		allErrs = append(allErrs, field.Invalid(fld, spec.LeaseDurationSeconds, "must be greater than 0"))
+	}
```

Comment:
I think you need to validate the version field(s) as semver.  If they get defaults they can be validated as non-empty, too.

I think you need to validate the TargetLease / LeaseName at least as non-empty, if not as a valid Lease name (not test if it exists, just that it COULD exist).

I think you need to validate RenewTime?

I think you need to validate the new Lease.strategy field (read https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/feature-gates.md for info about gates vs. validation).

Reply:
TODO: Validation is still WIP and needs to be addressed/reviewed until we finalize the API.